hazelcast/src/main/java/com/hazelcast/internal/nio/NodeIOService.java

```diff
@@ -73,74 +69,36 @@
-        restApiConfig = initRestApiConfig(node.getProperties(), node.getConfig());
-        memcacheProtocolConfig = initMemcacheProtocolConfig(node.getProperties(), node.getConfig());
+        restApiConfig = initRestApiConfig(node.getConfig());
+        memcacheProtocolConfig = initMemcacheProtocolConfig(node.getConfig());
     }
 
     /**
-     * Initializes {@link RestApiConfig} if not provided based on legacy group properties. Also checks (fails fast)
-     * if both the {@link RestApiConfig} and system properties are used.
+     * Initializes {@link RestApiConfig} if not provided based on legacy group properties.
      */
-    @SuppressWarnings("deprecation")
-    private static RestApiConfig initRestApiConfig(HazelcastProperties properties, Config config) {
-        boolean isAdvancedNetwork = config.getAdvancedNetworkConfig().isEnabled();
+    private static RestApiConfig initRestApiConfig(Config config) {
+        AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
+        boolean isAdvancedNetwork = advancedNetworkConfig.isEnabled();
         RestApiConfig restApiConfig = config.getNetworkConfig().getRestApiConfig();
-        boolean isRestConfigPresent = isAdvancedNetwork
-                ? config.getAdvancedNetworkConfig().getEndpointConfigs().get(EndpointQualifier.REST) != null
-                : restApiConfig != null;
-
-        if (isRestConfigPresent) {
-            // ensure the legacy Hazelcast group properties are not provided
-            ensurePropertyNotConfigured(properties, GroupProperty.REST_ENABLED);
-            ensurePropertyNotConfigured(properties, GroupProperty.HTTP_HEALTHCHECK_ENABLED);
-        }
-
-        if (isRestConfigPresent && isAdvancedNetwork) {
-            restApiConfig = new RestApiConfig();
-            restApiConfig.setEnabled(true);
-
-            RestServerEndpointConfig restServerEndpointConfig = config.getAdvancedNetworkConfig().getRestEndpointConfig();
-            restApiConfig.setEnabledGroups(restServerEndpointConfig.getEnabledGroups());
 
-        } else if (!isRestConfigPresent) {
-            restApiConfig = new RestApiConfig();
-            if (checkAndLogPropertyDeprecated(properties, GroupProperty.REST_ENABLED)) {
-                restApiConfig.setEnabled(true);
-                restApiConfig.enableAllGroups();
-            }
-            if (checkAndLogPropertyDeprecated(properties, GroupProperty.HTTP_HEALTHCHECK_ENABLED)) {
-                restApiConfig.setEnabled(true);
-                restApiConfig.enableGroups(RestEndpointGroup.HEALTH_CHECK);
-            }
+        if (isAdvancedNetwork && advancedNetworkConfig.getEndpointConfigs().get(EndpointQualifier.REST) != null) {
+            RestServerEndpointConfig restServerEndpointConfig = advancedNetworkConfig.getRestEndpointConfig();
+            restApiConfig.setEnabled(true).setEnabledGroups(restServerEndpointConfig.getEnabledGroups());
         }
+        restApiConfig.enableAllGroups();
```

Comment:
This line should be removed, we don't want all groups to be enabled always.

Reply:
After removing the line `WanRESTTest` tests start failing. Not sure what the cause, I'll try to find that out

Reply:
`restApiConfig#setEnabled(true)` makes the REST API config enabled with the default-on endpoint groups `CLUSTER_READ` and `HEALTH_CHECK`. In `WANRestTest` we also need to `enableGroups(RestEndpointGroup.WAN)` in `WANRestTest#getConfig()` since it uses the WAN group endpoints.

Reply:
Thank you. That helps. I pushed updated code.